src/cmatrix.cpp

```diff
@@ -1021,366 +1543,366 @@ void ImageMatrix::GetColorStatistics(double *hue_avg_p, double *hue_std_p, doubl
-// 	Map<Array< double, Dynamic, Dynamic, ColMajor >, Aligned > in_m (&in[0],height,width);
-// 	in_m = pix_plane;
-// 
-// 	fftw_execute(p);
-// 
-// 	// The resultant image uses the modulus (sqrt(norm)) of the complex numbers for pixel values
-// 	// Map an Eigen matrix onto what was returned by FFTW.
-// 	// Tricky bits specific to FFTW output:
-// 	//  We are making two matrixes out the single output - one for the real component and one for complex.
-// 	//  We are using strides to skip every other element of the output, starting at 0 for real, and 1 for complex.
-// 	//  We are not using the bottom half, making it symmetrical around the horizontal midline instead, but we still stride over twice the height b/c its complex
-// 	Map<const Array< double, Dynamic, Dynamic, ColMajor >, Aligned, Stride<Dynamic,2> > out_r (&out[0][0],half_height,width, Stride<Dynamic,2>(height*sizeof(fftw_complex),2));
-// 	Map<const Array< double, Dynamic, Dynamic, ColMajor >, Aligned, Stride<Dynamic,2> > out_i (&out[0][1],half_height,width, Stride<Dynamic,2>(height*sizeof(fftw_complex),2));
-// 	pix_plane.topRows(0,0,half_height,width) = (out_r.pow(2)+out_i.pow(2)).sqrt();
-// 
-// 	// complete the first column
-// 	unsigned int  odd_pad = (height % 2 == 0 ? 0 : 1);
-// 	pix_plane.block (half_height,0,half_height-odd_pad,1) = pix_plane.block (odd_pad,0,half_height-odd_pad,1).reverse();
-// 
-// 	// complete the rest of the columns
-// 	pix_plane.block (half_height,1,half_height-odd_pad,width-1) = pix_plane.block (odd_pad,1,half_height-odd_pad,width-1).reverse();
-// 	has_median = false;
-// 	stats.reset();
-// 
-// 	// clean up
-// 	fftw_destroy_plan(p);
-// 	fftw_free(in);
-// 	fftw_free(out);
-
-	return(0);
+    fftw_complex *out = (fftw_complex*) fftw_malloc(sizeof(fftw_complex) * width*height);
+    p = fftw_plan_dft_r2c_2d(width,height,in,out, FFTW_MEASURE); // FFTW_ESTIMATE: deterministic
+
+    unsigned int x,y;
+    for (x=0;x<width;x++)
+        for (y=0;y<height;y++)
+            in[height*x+y]=in_plane.coeff(y,x);
+
+    fftw_execute(p);
+
+    // The resultant image uses the modulus (sqrt(nrm)) of the complex numbers for pixel values
+    unsigned long idx;
+    for (x=0;x<width;x++) {
+        for (y=0;y<half_height;y++) {
+            idx = half_height*x+y;
+            out_plane (y,x) = stats.add (sqrt( pow( out[idx][0],2)+pow(out[idx][1],2)));    // sqrt(real(X).^2 + imag(X).^2)
+        }
+    }
+
+    // complete the first column
+    for (y=half_height;y<height;y++)
+        out_plane (y,0) = stats.add (out_plane (height - y, 0));
+
+    // complete the rest of the columns
+    for (y=half_height;y<height;y++)
+        for (x=1;x<width;x++)   // 1 because the first column is already completed
+            out_plane (y,x) = stats.add (out_plane (height - y, width - x));
+
+    // clean up
+    fftw_destroy_plan(p);
+    fftw_free(in);
+    fftw_free(out);
+
+    //
```

Comment:
Please don't leave large chunks of uncommented code. If you think it's important then leave a comment. Otherwise we should get rid of it.

Reply:
Nothing here is my contribution. I just ran Auto-Indent for the code here